File: app/src/main/java/com/jerbo/llanteria/Clientes/AddCliente.java
package com.jerbo.llanteria.Clientes;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;
import com.google.android.material.textfield.TextInputLayout;
import androidx.appcompat.app.AppCompatActivity;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;
import com.jerbo.llanteria.PaginaPrincipal.MainActivity;
import com.jerbo.llanteria.R;

public class AddCliente extends AppCompatActivity {
    TextInputLayout nombre, apellido, direccion, correo, dni;
    Button aceptar, cancelar;
    ProgressBar progressBar;
    TextView test;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_cliente);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        createIU();
    }

    private void createIU() {
        nombre = findViewById(R.id.nombre_cliente);
        apellido = findViewById(R.id.apellidos_cliente);
        direccion = findViewById(R.id.direccion_cliente);
        correo = findViewById(R.id.email_cliente);
        dni = findViewById(R.id.dni_cliente);
        aceptar = findViewById(R.id.aceptar_button_cliente);
        cancelar = findViewById(R.id.cancelar_button_cliente);
        progressBar = findViewById(R.id.cliente_progress_bar);
        test = findViewById(R.id.test);
        aceptar.setOnClickListener(v -> {
            if (!checkData()) {
                createUser();
            }
        });
        cancelar.setOnClickListener(v -> {
            salir();
        });
    }
    public void salir(){
        AlertDialog.Builder builder = new AlertDialog.Builder(this);

        builder.setMessage("Desea cancelar la operacion actual?")
                .setCancelable(false)
                .setPositiveButton("Si", (dialog, which) -> {
                    startActivity(new Intent(this,MainActivity.class));
                })
                .setNegativeButton("No", (dialog, which) -> dialog.cancel());
        AlertDialog alertDialog = builder.create();
        alertDialog.show();
    }

    private void createUser() {
        displayProgressBar(true);
        RequestQueue queue = Volley.newRequestQueue(this);
        String url = "https://llanteriamari.000webhostapp.com/insert_user.php?" +
                "p1=" + dni.getEditText().getText().toString().trim() + " & " +
                "p2=" + nombre.getEditText().getText().toString().trim() + " & " +
                "p3=" + apellido.getEditText().getText().toString().trim() + " & " +
                "p4=" + direccion.getEditText().getText().toString().trim() + " & " +
                "p5=" + correo.getEditText().getText().toString().trim();
        StringRequest stringRequest = new StringRequest(Request.Method.GET, url, response -> {
            if (response.equals("1")) {
                displayProgressBar(false);
                Toast.makeText(this, "Usuario añadido exitosamente", Toast.LENGTH_SHORT).show();
            } else {
                displayProgressBar(false);
                Toast.makeText(this, "Ocurrio un error", Toast.LENGTH_SHORT).show();
            }

        }, error -> {

        });
        queue.add(stringRequest);
    }

    private boolean checkData() {
        if (!validateNombre() | !validateApellido() | !validateDni() | !validateCorreo() | !validateDireccion()) {
            return true;
        }else
            return false;

    }

    private boolean validateCorreo() {
        String t = correo.getEditText().getText().toString().trim();
        if (!t.isEmpty() && t.matches("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")) {
            correo.setError(null);
            return true;
        } else {
            correo.setError("Ingrese correo valido");
            return false;

        }
    }

    private boolean validateDireccion() {
        String t = direccion.getEditText().getText().toString().trim();
        if (t.isEmpty()) {
            direccion.setError("Ingrese direccion valida");
            return false;
        } else {
            direccion.setError(null);
            return true;
        }
    }

    private boolean validateDni() {
        String t = dni.getEditText().getText().toString().trim();
        if (!t.isEmpty() && t.length() == 8) {
            dni.setError(null);
            return true;
        } else {
            dni.setError("ingrese un DNI valido");
            return false;
        }
    }

    private boolean validateApellido() {
        String t = apellido.getEditText().getText().toString().trim();
        if (t.isEmpty()) {
            apellido.setError("Ingrese un apellido");
            return false;
        } else {
            apellido.setError(null);
            return true;
        }
    }

    private boolean validateNombre() {
        String t = nombre.getEditText().getText().toString().trim();
        if (t.isEmpty()) {
            nombre.setError("Ingrese un nombre");
            return false;
        } else {
            nombre.setError(null);
            return true;
        }
    }

    private void displayProgressBar(boolean b)  {
        if (b) {
            nombre.setVisibility(View.GONE);
            apellido.setVisibility(View.GONE);
            dni.setVisibility(View.GONE);
            direccion.setVisibility(View.GONE);
            correo.setVisibility(View.GONE);
            aceptar.setVisibility(View.GONE);
            cancelar.setVisibility(View.GONE);
            progressBar.setVisibility(View.VISIBLE);
        } else {
            nombre.setVisibility(View.VISIBLE);
            apellido.setVisibility(View.VISIBLE);
            dni.setVisibility(View.VISIBLE);
            direccion.setVisibility(View.VISIBLE);
            correo.setVisibility(View.VISIBLE);
            aceptar.setVisibility(View.VISIBLE);
            cancelar.setVisibility(View.VISIBLE);
            progressBar.setVisibility(View.GONE);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                salir();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    public void onBackPressed() {
        salir();
    }
}

File: app/src/main/java/com/jerbo/llanteria/Clientes/AddCliente.kt
package com.jerbo.llanteria.Clientes

import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import com.jerbo.llanteria.PaginaPrincipal.MainActivity
import com.jerbo.llanteria.R
import kotlinx.android.synthetic.main.activity_add_cliente.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlinx.coroutines.launch
import kotlin.coroutines.CoroutineContext
import kotlin.coroutines.resume
import kotlin.coroutines.suspendCoroutine

open class AddCliente : AppCompatActivity(), CoroutineScope {
    private lateinit var job: Job
    override val coroutineContext: CoroutineContext
        get() = job + Dispatchers.Main
    private lateinit var queue: RequestQueue

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_cliente)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        createIU()
    }

    private fun createIU() {
        job = Job()
        queue = Volley.newRequestQueue(this)
        aceptar_button_cliente.setOnClickListener {
            if (!checkData()) {
                displayProgressBar(true)
                launch {
                    if (checkIfUserExists())
                        createUser()
                    else {
                        displayProgressBar(false)
                        Toast.makeText(this@AddCliente, "Usuario ya registrado con ese DNI", Toast.LENGTH_SHORT).show()
                    }
                }


            }
        }
        cancelar_button_cliente.setOnClickListener { salir() }
    }

    suspend fun checkIfUserExists() = suspendCoroutine<Boolean> { cont ->
        val url =
            "https://llanteriamari.000webhostapp.com/checkDni.php?p1=" + dni_cliente.editText!!.text.toString().trim { it <= ' ' }
        val stringRequest = StringRequest(Request.Method.GET, url,
            { response ->
                cont.resume(response.trim() == "0")
                Log.e(TAG, "checkIfUserExists: $response")
            },
            { })
        queue.add(stringRequest)

    }

    private fun salir() {
        val builder = AlertDialog.Builder(this)
        builder.setMessage("Desea cancelar la operacion actual?")
            .setCancelable(false)
            .setPositiveButton("Si") { _, _ ->
                run {
                    startActivity(Intent(this, MainActivity::class.java))
                    super.onBackPressed()
                }
            }
            .setNegativeButton("No") { dialog, _ -> dialog.cancel() }
        val alertDialog = builder.create()
        alertDialog.show()
    }

    private fun createUser() {


        val url = "https://llanteriamari.000webhostapp.com/insert_user.php?" +
                "p1=" + dni_cliente.editText!!.text.toString().trim { it <= ' ' } + " & " +
                "p2=" + nombre_cliente.editText!!.text.toString().trim { it <= ' ' } + " & " +
                "p3=" + apellidos_cliente.editText!!.text.toString().trim { it <= ' ' } + " & " +
                "p4=" + direccion_cliente.editText!!.text.toString().trim { it <= ' ' } + " & " +
                "p5=" + email_cliente.editText!!.text.toString().trim { it <= ' ' }
        val stringRequest = StringRequest(Request.Method.GET, url, { response ->
            if (response == "1") {
                displayProgressBar(false)
                Toast.makeText(this, "Usuario añadido exitosamente", Toast.LENGTH_SHORT).show()
                dni_cliente.editText!!.setText("")
                nombre_cliente.editText!!.setText("")
                apellidos_cliente.editText!!.setText("")
                direccion_cliente.editText!!.setText("")
                email_cliente.editText!!.setText("")
            } else {
                displayProgressBar(false)
                Toast.makeText(this, "Ocurrio un error", Toast.LENGTH_SHORT).show()
            }

        }, {})
        queue.add(stringRequest)
    }

    private fun checkData(): Boolean {
        return !validateNombre() or !validateApellido() or !validateDni() or !validateCorreo() or !validateDireccion()

    }

    private fun validateCorreo(): Boolean {
        val t = email_cliente.editText!!.text.toString().trim { it <= ' ' }
        return if (t.isNotEmpty() && t.matches("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$".toRegex())) {
            email_cliente.error = null
            true
        } else {
            email_cliente.error = "Ingrese correo valido"
            false

        }
    }

    private fun validateDireccion(): Boolean {
        val t = direccion_cliente.editText!!.text.toString().trim { it <= ' ' }
        return if (t.isEmpty()) {
            direccion_cliente.error = "Ingrese direccion valida"
            false
        } else {
            direccion_cliente.error = null
            true
        }
    }

    private fun validateDni(): Boolean {
        val t = dni_cliente.editText!!.text.toString().trim { it <= ' ' }
        return if (t.isNotEmpty() && t.length == 8) {
            dni_cliente.error = null
            true
        } else {
            dni_cliente.error = "ingrese un DNI valido"
            false
        }
    }

    private fun validateApellido(): Boolean {
        val t = apellidos_cliente.editText!!.text.toString().trim { it <= ' ' }
        return if (t.isEmpty()) {
            apellidos_cliente.error = "Ingrese un apellido"
            false
        } else {
            apellidos_cliente.error = null
            true
        }
    }

    private fun validateNombre(): Boolean {
        val t = nombre_cliente.editText!!.text.toString().trim { it <= ' ' }
        return if (t.isEmpty()) {
            nombre_cliente.error = "Ingrese un nombre"
            false
        } else {
            nombre_cliente.error = null
            true
        }
    }

    private fun displayProgressBar(b: Boolean) {
        if (b) {
            nombre_cliente.visibility = View.GONE
            apellidos_cliente.visibility = View.GONE
            dni_cliente.visibility = View.GONE
            direccion_cliente.visibility = View.GONE
            email_cliente.visibility = View.GONE
            aceptar_button_cliente.visibility = View.GONE
            cancelar_button_cliente.visibility = View.GONE
            cliente_progress_bar.visibility = View.VISIBLE
        } else {
            nombre_cliente.visibility = View.VISIBLE
            apellidos_cliente.visibility = View.VISIBLE
            dni_cliente.visibility = View.VISIBLE
            direccion_cliente.visibility = View.VISIBLE
            email_cliente.visibility = View.VISIBLE
            aceptar_button_cliente.visibility = View.VISIBLE
            cancelar_button_cliente.visibility = View.VISIBLE
            cliente_progress_bar.visibility = View.GONE
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                salir()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onBackPressed() {
        salir()
    }

    companion object {
        private const val TAG = "AddClientejava"
    }
}
